io/net/net_tools.py

```diff
@@ -0,0 +1,374 @@
+        fobj.write(self.hostname)
+        fobj.close()
+        fobj = open(self.myhostname_file, 'w')
+        fobj.write("myhost.my-domain")
+        fobj.close()
+
+    def test_hostname(self):
+        """
+        i.Verifies hostname command options and
+        ii.Test to change the hostname
+        """
+        output = process.run("hostname", verbose=False,
+                             env={"LANG": "C"})
+        if not output.stdout:
+            self.error("unexpected response from hostname command")
+
+        # Verifying different options provided by hostname
+        options_to_verify = self.params.get('hostname_opt', default="f")
+        for option in options_to_verify:
+            output = process.run("hostname -%s" % option, verbose=False)
+            if not output.stdout:
+                self.error("unexpected response from hostname command")
+            self.log.info("verified -%s option of hostname command" % option)
+
+        # Test to change hostname
+        process.system("hostname -F %s" % self.myhostname_file,
+                       verbose=False,
+                       sudo=True)
+        self.restore_hostname = True
+        output = process.run("hostname", verbose=False,
+                             env={"LANG": "C"})
+        if "myhost.my-domain" not in output.stdout:
+            self.error("unexpected response from hostname -F command and " +
+                       "hostname -F didn't set hostname")
+        ret = self._restore_hostname(self.hostname_file, self.hostname)
+        if ret:
+            self.error(ret)
+        self.restore_hostname = False
+
+    @staticmethod
+    def _restore_hostname(hostnamefile, hostname):
+        # Restore hostname
+        process.system("hostname -F %s" % hostnamefile,
+                       verbose=False)
+        output = process.run("hostname", verbose=False,
+                             env={"LANG": "C"})
+        if hostname not in output.stdout:
+            return "Failed to restore Hostname"
+
+    def tearDown(self):
+        if self.restore_hostname:
+            ret = self._restore_hostname(self.hostname_file, self.hostname)
+            if ret:
+                self.error(ret)
+        shutil.rmtree(self.backup_dir)
+
+
+class Ifconfig(Test):
+    """
+    ifconfig - configure a network interface
+    Test to verify the ifconfig functionality.
+    """
+
+    def setUp(self):
+        if "0% packet loss" not in process.system_output(
+                "ping -c 5 -w 5 localhost", verbose=False):
+            self.skip("Loopback interface is in down state")
+
+        self.lo_intf_state = "Up"
+        if "lo:1" in process.system_output("ifconfig", verbose=False):
+            self.skip("alias for an loopback interface is configured")
+        self.alias = None
+        install_dependencies()
+        self.ipv6 = False
+        if os.path.exists("/proc/net/if_inet6"):
+            self.ipv6 = True
+
+    def test_ifconfig(self):
```

Comment:
All of the process exceptions are actually test failures, so please use the `fail_on` decorator here as well.